hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -259,14 +259,7 @@ public static void setColumnProjection(Configuration configuration, CarbonProjec
     if (projection == null || projection.isEmpty()) {
       return;
     }
-    String[] allColumns = projection.getAllColumns();
-    StringBuilder builder = new StringBuilder();
-    for (String column : allColumns) {
-      builder.append(column).append(",");
-    }
-    String columnString = builder.toString();
-    columnString = columnString.substring(0, columnString.length() - 1);
-    configuration.set(COLUMN_PROJECTION, columnString);
+    setColumnProjection(configuration, projection.getAllColumns());
```

Comment:
1.  Reverted the change (deserialized => deserialize)
2.  Removed
3.  If the configuration exists and not equal to 'true', it will be false
4.  Updated